app/controllers/energy_tariffs/energy_tariff_charges_controller.rb

```diff
@@ -6,11 +6,9 @@ def index
 
     def create
       @energy_tariff_charges = make_charges(params[:energy_tariff_charges], @energy_tariff)
-      @energy_tariff_charges.map(&:valid?)
       if @energy_tariff_charges.all?(&:valid?)
         EnergyTariff.transaction do
           @energy_tariff.update(energy_tariff_params)
-          @energy_tariff.energy_tariff_charges.destroy_all
```

Comment:
This is the create method in the energy tariff charges controller, so the energy tariff will already exist. And it may have existing charges. Currently we remove any existing charges and replace with the current form state. So this is required. Otherwise we'll need to look at upserting each individual charge in the collection.